open-metadata-resources/open-metadata-labs/ui-labs/ui-asset-search.ipynb

```diff
@@ -0,0 +1,290 @@
+    "\n",
+    "For this part of the exercises, we have prepared a set of events for Data Engine OMAS in `cocoMDS1` to pick up. The precise process on how these events have been created is more comvoluted, and is out of scope of this lab."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Install kafka-python"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will send the IGC events through the kafka bus. The `kafka-python` package allows us to interact with kafka directly with Python."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install kafka-python"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Import IGC events"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Before importing the events, we first pull the value from an environment variable called `eventBusURLroot` with a default value of `localhost:9092`.  It is used in all of the server configuration documents to connect it to Kafka."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "eventBusURLroot = os.environ.get('eventBusURLroot', 'localhost:9092')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The IGC events to create assets are located in the text file `igc-events.txt`. One line represents one event. The kafka topic we should send these events to is `egeria.omag.openmetadata.repositoryservices.cohort.cocoCohort.OMRSTopic`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from kafka import KafkaProducer, KafkaClient\n",
+    "\n",
+    "producer = KafkaProducer(bootstrap_servers=eventBusURLroot)\n",
+    "\n",
+    "with open('igc-events.txt', 'r') as f:\n",
+    "    for line in f:\n",
+    "        producer.send(\n",
+    "            'egeria.omag.openmetadata.repositoryservices.cohort.cocoCohort.OMRSTopic', \n",
+    "            str.encode(line)\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The code above reads one event at a time, and sends it to the `egeria.omag.openmetadata.repositoryservices.cohort.cocoCohort.OMRSTopic` kafka topic. All OMAS servers listening to this topic will attempt to pick up and process these events. Once successful, we can search for the assets in the UI."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Exercise 2\n",
+    "\n",
+    "### Search for assets in the UI"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can go to the UI, located at:\n",
+    "```\n",
+    "https://localhost:18443\n",
```

Comment:
Not necessarily. It will be be
 * https://localhost:8443 locally
 * https://datalake:8443 OR https://localhost:18443 in compose
 * https://<one-of-my-k8s-nodes>:30443 in k8s

We may need to ensure we inject the right environment variables into the jupyter container so that it can display the correct value (if we can't determine it within the notebook)

Reply:
Added in 2583e0528dbb039feddda24889f9e5769381f449.